api/src/Modules/staff/Handler/StaffRBAC.php

```diff
@@ -85,9 +85,7 @@ public function install($container)
                 $new = $value['DepartmentID'].'.'.$pos;
                 $this->addRole($new, [$parent]);
                 $this->addToAll('all.staff', $new, $pos);
-                if ($parent == $firstparent) {
-                    $parent = $new;
-                }
+                $parent = $new;
```

Comment:
I think this only resolves part of the issue with RBAC in this implementation.

I am seeing the following behavior.

Division Director and Division Support seem to have the correct division permissions. In addition, they both seem to have things that would have been added with `all.staff`.

For Department Heads, Subheads, and Specialists, they seem to have the correct department permissions. However they're all missing permissions that _should have_ been added with `all.staff` role, such as `api.get.staff`, `api.get.announcement.staff`, `api.get.deadline.staff`, `api.get.volunteer.hours` and `api.post.volunteer.hours`.

Previously, I was not seeing any permissions for Department Subheads, so that appears to be resolved. But these others are still missing - is there something I'm not doing correctly in testing? I've been calling up the member permission API endpoint to verify the changes here.

Reply:
I think I was able to track this down. In addition to your change here, it looks like there is maybe one more change required.

`api/src/Vendors/Zend/ZendRBACBackend.php` in `addRoleParents()` function, change `$r->addParent($p);` on line 126 to `$p->addChild($r);`

For whatever reason, I think the roles were not being associated correctly and did not have the correct number of children. I just happened to try this because I wanted to see if reversing the relationship would work, and it did. Couldn't tell you why, but thought it was worth a try.

Here is some sample log output indicating how I suspected and confirmed this:

Department SubHead should at least have `all.staff`, `all.2` and `117.3` in this case for this Dept. But they didn't have anything.
```
2023-10-27 21:21:39 [28-Oct-2023 02:21:39] WARNING: [pool www] child 7 said into stderr: "NOTICE: PHP message: In getPermissions() for 117.2 and 1"
2023-10-27 21:21:39 [28-Oct-2023 02:21:39] WARNING: [pool www] child 7 said into stderr: "NOTICE: PHP message: Stored role: 117.2"
2023-10-27 21:21:39 [28-Oct-2023 02:21:39] WARNING: [pool www] child 7 said into stderr: "NOTICE: PHP message: Length of children for role: 0"
```

When I made the switch, I got this output instead:
```
2023-10-27 21:24:39 [28-Oct-2023 02:24:39] WARNING: [pool www] child 7 said into stderr: "NOTICE: PHP message: In getPermissions() for 117.2 and 1"
2023-10-27 21:24:39 [28-Oct-2023 02:24:39] WARNING: [pool www] child 7 said into stderr: "NOTICE: PHP message: Stored role: 117.2"
2023-10-27 21:24:39 [28-Oct-2023 02:24:39] WARNING: [pool www] child 7 said into stderr: "NOTICE: PHP message: Length of children for role: 3"
2023-10-27 21:24:39 [28-Oct-2023 02:24:39] WARNING: [pool www] child 7 said into stderr: "NOTICE: PHP message: Role child: all.staff"
2023-10-27 21:24:39 [28-Oct-2023 02:24:39] WARNING: [pool www] child 7 said into stderr: "NOTICE: PHP message: Role child: all.2"
2023-10-27 21:24:39 [28-Oct-2023 02:24:39] WARNING: [pool www] child 7 said into stderr: "NOTICE: PHP message: Role child: 117.3"
```